datafusion/core/tests/sql/timestamp.rs

```diff
@@ -266,6 +266,37 @@ async fn query_cast_timestamp_micros_to_others() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn query_cast_timestamp_from_unixtime() -> Result<()> {
+    let ctx = SessionContext::new();
+
+    let t1_schema = Arc::new(Schema::new(vec![Field::new("ts", DataType::Int64, true)]));
+    let t1_data = RecordBatch::try_new(
+        t1_schema.clone(),
+        vec![Arc::new(Int64Array::from(vec![
+            1235865600, 1235865660, 1238544000,
+        ]))],
+    )?;
+    let t1_table = MemTable::try_new(t1_schema, vec![vec![t1_data]])?;
+    ctx.register_table("t1", Arc::new(t1_table))?;
+
+    let sql = "SELECT from_unixtime(ts) FROM t1 LIMIT 3";
+    let actual = execute_to_batches(&ctx, sql).await;
+
+    let expected = vec![
```

Comment:
For what it is worth `mysql` gives a different answer (defaults to local time)

```sql
mysql> select from_unixtime(1238544000);
+---------------------------+
| from_unixtime(1238544000) |
+---------------------------+
| 2009-03-31 20:00:00       |
+---------------------------+
1 row in set (0.00 sec)
```

However, when I set the session timezone to UTC things look good
```sql
mysql> SET time_zone = '+0:00';
Query OK, 0 rows affected (0.00 sec)

mysql> select from_unixtime(1238544000);
+---------------------------+
| from_unixtime(1238544000) |
+---------------------------+
| 2009-04-01 00:00:00       |
+---------------------------+
1 row in set (0.00 sec)

```


(This behavior is not unique to `from_unixtime` I think it affects all timestamp arithmetic in DataFusion)